plugin/metadata/metadata.go

```diff
@@ -0,0 +1,48 @@
+package metadata
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+// Metadater interface needs to be implemented by each plugin willing to provide
+// metadata information for other plugins.
+// Note: this method should work quickly, because it is called for every request
+// from the metadata plugin.
+type Metadater interface {
+	// Metadata gets content, ResponseWriter and dns.Msg and returns context with
+	// additional values. Metadata must be thread safe.
+	Metadata(context.Context, dns.ResponseWriter, *dns.Msg) (context.Context, error)
+}
+
+// Metadata implements collecting metadata information from all enabled plugins
+// which provide it
```

Comment:
// Metadata implements collecting metadata information from all plugins that implement the Metadater interface.
￼